MSVC 2015 Compiler crash when building HPX

On revision 5609f1cddb335fb1f MSVC 2015 with Update 3 (v14.0.25431.01) crashes when building HPX:

```
hpx\src\runtime\agas\big_boot_barrier.cpp(289): fatal error C1001: An internal error has occurred in the compiler.
```

Probably this is related to the `operator&` in file [`src/runtime/agas/big_boot_barrier.cpp`](https://github.com/STEllAR-GROUP/hpx/blame/master/src/runtime/agas/big_boot_barrier.cpp) and the following lines:

- `ar & typenames`
- `ar & num_localities`
- `ar & used_cores`

```cpp
    template <typename Archive>
    void serialize(Archive & ar, const unsigned int)
    {
        ar & endpoints;
        ar & primary_ns_ptr;
        ar & symbol_ns_ptr;
        ar & cores_needed;
        ar & num_threads;
        ar & hostname;
        ar & typenames;
     // ^^^^^^^^^^^^^^^
        ar & prefix;
    }
```
... and ...

```cpp
    template <typename Archive>
    void serialize(Archive & ar, const unsigned int)
    {
        ar & prefix;
        ar & agas_locality;
        ar & locality_ns_address;
        ar & primary_ns_address;
        ar & component_ns_address;
        ar & symbol_ns_address;
        ar & num_localities;
     // ^^^^^^^^^^^^^^^
        ar & used_cores;
     // ^^^^^^^^^^^^^^^
        ar & agas_endpoints;
        ar & ids;
        ar & endpoints;
    }
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1081,7 +1081,11 @@ endif()
 if(WIN32)
   if(MSVC)
     enable_language(ASM_MASM)
+
     hpx_add_target_compile_option(-Ox CONFIGURATIONS Release)
+
+    # even VS2017 has an ICE when compiling with -Ob2
+    hpx_add_target_compile_option(-Ob1 CONFIGURATIONS Release)
 
     if(NOT HPX_WITH_AWAIT)
       # /RTC1 is incompatible with /await

